### DISCUSSION TITLE
Is GO install required?

### DISCUSSION THREAD

**Discussion by dsaldivar** [danielmiessler/Fabric#1176] [General]
Hello,
I haven't revisited my fabric Python installation I did in the first of the year until now.  In reviewing the current installation instructions it indicates that fabric SHOULD be installed using GO.  My question is:  Is it NECESSARY to upgrade fabric by reinstalling it using the GO installation instructions?  Can I get an upgraded fabric build while staying on the Python version?  The reason I'm wanting to stay on the Python version is because I've gone through some of the discussions and there appears to be some problematic issues after making the transition from the Python version to the GO version.  Just wondering if support for the Python will be ongoing or will the transition to the GO version be required to get any further updates to fabric.   Thanks in advance for your feedback. Danny

**Comment by bledburn** [comment]
> Hello, I haven't revisited my fabric Python installation I did in the first of the year until now. In reviewing the current installation instructions it indicates that fabric SHOULD be installed using GO. My question is: Is it NECESSARY to upgrade fabric by reinstalling it using the GO installation instructions? Can I get an upgraded fabric build while staying on the Python version? The reason I'm wanting to stay on the Python version is because I've gone through some of the discussions and there appears to be some problematic issues after making the transition from the Python version to the GO version. Just wondering if support for the Python will be ongoing or will the transition to the GO version be required to get any further updates to fabric. Thanks in advance for your feedback. Danny

The Python version of Fabric is now deprecated. The new Go version is now probably more stable anyhow, and it certainly has more features for many, many more use cases. You can just install the prebuilt binary (like a .exe file for the Windows user's out there), you can install using Go, and now you can also install with a package manager (Homebrew/Linuxbrew and the AUR)

For installing just the binary/executable, watch here: https://youtu.be/hnFJhubYS88

For installing with Go, watch here: https://youtu.be/_QMYQtYUpys

And to install with a package manager, watch here: https://youtu.be/hMaPGbitMww

The package manager demonstration uses Brew, not the AUR, as I was demonstrating Fabric on Ubuntu Linux, not Arch Linux.

**Reply by ksylvan** [reply]
The new one line installation method is superfluous since simple and cross-platform:

https://github.com/danielmiessler/Fabric/discussions/1746
